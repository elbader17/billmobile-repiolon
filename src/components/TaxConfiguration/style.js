import { StyleSheet } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../constants/colors';
import { FONTS } from '../../constants/fonts';

const styles = StyleSheet.create({
  container: {
    height: hp('100%')
  },
  containerBody: {
    flex: 0.50,
    marginHorizontal: 25,
    alignItems: 'center',
    justifyContent: 'center'
  }, 
  containerFooter: {
    flex: 0.50,
    alignItems: 'center',
  },
  containerInput: {
    marginHorizontal: 25,
    marginTop: 10,
  },
  image: {
    width: 360,
    height: 225,
  },
  center: {
    marginTop: 15,
    alignItems: 'center'
  },
  button: {
    width: wp('65%'),
    height: hp('7%'),
    backgroundColor: 'transparent',
    borderRadius: 25,
    elevation: 1,
    marginTop: 15,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textRegular18Blue: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    color: COLORS.blueMedium,
    textAlign: 'center'
  },
  textRegular14GrayDark: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.grayDark,
    textAlign: 'center',
    marginTop: 10
  },
  textRegular14Gray: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.gray,
    textAlign: 'center',
  },
  textRegular14white: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.white,
    textAlign: 'center'
  },
  headerText: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    fontWeight: 'normal',
    color: COLORS.white,
  }
});

export default styles;