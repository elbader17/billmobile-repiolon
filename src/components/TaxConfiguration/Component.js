import React from 'react';
import { View, Text, TextInput, Alert } from 'react-native';
import { Button } from "react-native-elements";
import { withNavigation } from 'react-navigation';
import style from './style';

class TaxConfiguration extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
      name: '',
      cuit: '',
      keyfiscal:'',
    };
  }

  handleConfigFiscal = () => {
   
   Alert.alert("Towken: "+this.props.jwtToken);
   const { name, cuit, keyfiscal } = this.state;
   const { registerUserService } = this.props;
   registerUserService(name, cuit, keyfiscal)
    .then((data) => {
      Alert.alert("Pops Data: "+this.props.name+" "+this.props.cuit);
      this.props.navigation.navigate('HomeScreen');
    })
  }

  setName = (value) => this.setState({ name: value})
  setCuit = (value) => this.setState({ cuit: value })
  setKeyFiscal = (value) => this.setState({ keyfiscal: value })
    
  render() {
    return(
      <View style={style.container}>
        <View style={style.container2}>
          <View style={ style.textBoxBtnHolder }>
            <Text style={ style.textRegister }>
              NOMBRE DE LA EMPRESA
            </Text>
            <TextInput style={ style.textRegister }
              onChangeText={this.setName}
              style={ style.textBox }
            />
            <Text style={ style.textRegister }>
              INGRESA TU CUIT
            </Text>
            <TextInput style={ style.textRegister }
              onChangeText={this.setCuit}
              style={ style.textBox }
            />
            <Text style={ style.textRegister }>
              POR AHORA SIN CLAVE FISCAL
            </Text>
            <Button
              title="LISTO"
              onPress={ this.handleConfigFiscal }
              buttonStyle={ style.submit }
              titleStyle={ style.submitText }
              disabledTitleStyle={ style.submitText }
              disabledStyle={ style.submitDisabled }
            />
          </View>
        </View>
      </View>
    )
  }
}

export default withNavigation(TaxConfiguration);