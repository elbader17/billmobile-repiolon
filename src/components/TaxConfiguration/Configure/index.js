import { connect } from 'react-redux';
import TaxConfiguration from './Component';
import { updateFiscalIdentity, getCertificate, uploadCertificate } from '../../../app/user_service/actions';

const mapStateToProps = state => ({
  name: state.userservice.name,
  cuit: state.userservice.cuit,
});

function mapDispatchToProps(dispatch) {
  return {
    updateFiscalIdentity: (name, cuit) => dispatch(updateFiscalIdentity(name, cuit)),
    getCertificate: () => dispatch(getCertificate()),
    updateCertificate: (pkey, cert) => dispatch(uploadCertificate(pkey, cert))
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(TaxConfiguration);

export default component;
