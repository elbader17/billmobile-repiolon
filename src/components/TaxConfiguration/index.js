import { connect } from 'react-redux'
import TaxConfiguration from './component';
import { registerUserService } from '../../app/userservice/action';

const mapStateToProps = state => ({
  jwtToken: state.authentication.jwtToken,
  name: state.userservice.name,
  cuit: state.userservice.cuit,
  keyfiscal: state.userservice.keyfiscal,
});

function mapDispatchToProps(dispatch) {
  return {
    registerUserService: (name, cuit, clavefiscal) => dispatch(registerUserService(name, cuit, clavefiscal)),
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(TaxConfiguration);

export default component;
