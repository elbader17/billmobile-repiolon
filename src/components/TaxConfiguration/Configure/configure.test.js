import React from 'react';
import { shallow } from 'enzyme';
import TaxConfiguration from './Component';

jest.mock('../../../constants/metrics', () => ({ METRICS: { heightHeader: {} } }));

const promise = Promise.resolve(0);
const updateFiscalIdentityMock = jest.fn(() => promise);

test('render correctly component TaxConfiguration', () => {
  const component = shallow(<TaxConfiguration />)
  expect(component).toMatchSnapshot();
});

test('call the createItem function with param product', () => {
  const component = shallow(<TaxConfiguration updateFiscalIdentity={updateFiscalIdentityMock}/>);
  const button = component.findWhere(node => node.prop('testID') === 'ready');
  button.simulate('press');
  expect(updateFiscalIdentityMock).toBeCalled();
});

describe('check disabled button ready', () => {
  test('button disabled at startup', () => {
    const component = shallow(<TaxConfiguration />);
    const button = component.findWhere(node => node.prop('testID') === 'ready');
    expect(button.props().disabled).toBe(true);
  });

  test('LISTO button enabled with the correct data', () => {
    const component = shallow(<TaxConfiguration />);
    component.setState({name:'Martin', cuit:'20363095721'});
    const button = component.findWhere(node => node.prop('testID') === 'ready');
    expect(button.props().disabled).toBe(false);
  });

    /*
    test('LISTO button disabled with name empty', () => {
      const component = shallow(<TaxConfiguration />);
      component.setState({name:'', cuit:'20363095721'});
      const button = component.findWhere(node => node.prop('testID') === 'ready');
      expect(button.props().disabled).toBe(false);
    });
    */

  test('LISTO button disabled with cuit invalid', () => {
    const component = shallow(<TaxConfiguration />);
    component.setState({name:'Martin', cuit:'0363095721'});
    const button = component.findWhere(node => node.prop('testID') === 'ready');
    expect(button.props().disabled).toBe(true);
  });
});

test('check state loading', () => {
  const component = shallow(<TaxConfiguration updateFiscalIdentity={updateFiscalIdentityMock} />);
  expect(component.instance().state.loading).toBe(false);
  const button = component.findWhere(node => node.prop('testID') === 'ready');
  button.simulate('press');
  expect(component.instance().state.loading).toBe(true);
});