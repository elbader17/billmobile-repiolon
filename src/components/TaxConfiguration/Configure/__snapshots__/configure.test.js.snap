// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly component TaxConfiguration 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={Object {}}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollViewMock>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": NaN,
          "paddingHorizontal": 18,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "justifyContent": "center",
            "marginTop": 5,
            "position": "relative",
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#858585",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 15,
                  "textAlign": "center",
                },
                Object {
                  "paddingVertical": 5,
                },
              ]
            }
          >
            NOMBRE DE LA EMPRESA
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "#C4C4C4",
                "borderRadius": 2,
                "borderWidth": 1,
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 18,
                "height": 45,
                "paddingLeft": 8,
                "paddingRight": 45,
                "paddingVertical": 0,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#EEEEEE",
              "height": 1,
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        />
        <View
          style={Object {}}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#858585",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 15,
                  "textAlign": "center",
                },
                Object {
                  "paddingVertical": 5,
                },
              ]
            }
          >
            INGRESA TU CUIT
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            onEndEditing={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "#C4C4C4",
                "borderRadius": 2,
                "borderWidth": 1,
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 18,
                "height": 45,
                "paddingLeft": 8,
                "paddingRight": 45,
                "paddingVertical": 0,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#EEEEEE",
              "height": 1,
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        />
      </View>
    </View>
  </ScrollViewMock>
  <ForwardRef(Themed.Button)
    buttonStyle={
      Object {
        "backgroundColor": "#EE6123",
        "borderRadius": 0,
        "height": 106.5,
      }
    }
    disabled={true}
    disabledStyle={
      Object {
        "backgroundColor": "#C4C4C4",
        "borderRadius": 0,
        "height": 106.5,
      }
    }
    disabledTitleStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Lato-Semibold",
        "fontSize": 14,
      }
    }
    loading={false}
    onPress={[Function]}
    testID="ready"
    title="LISTO"
    titleStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Lato-Semibold",
        "fontSize": 14,
      }
    }
  />
</KeyboardAvoidingView>
`;
